src/Samples/CodeEditorContribution/Contribution.d.ts

```diff
@@ -0,0 +1,340 @@
+import { GitVersionDescriptor, TfvcVersionDescriptor } from "azure-devops-extension-api/Git/Git";
+
+/**
+ * Extensions to the code editor should target this extension point.
+ */
+export type ExtensionPoint = "ms.vss-code-web.code-editor-contribution-collection";
+
+/**
+ * Extensions to the code editor should implement this interface, which will be called when the exension is loaded.
+ */
+export interface ICodeEditorContribution {

```

Comment:
Are we following whatever VS Code exposes? Since that eco system is already so large, it might be good to make it as easy as possible to reuse

Reply:
Yes and no. 

ILanguageExtensionPoint, IJsonSchemaRegistration, and LanguageConfiguration are all identical to VS Code equivalents (based on contributes.languages and vscode-json-languageservice).

IMonarchLanguage is from Monaco, which is the core editor of VS Code repackaged for web browsers (and the code editor on Azure DevOps). From what I understand, the simplified explanation for why this isn't the same as VS Code is that VS Code uses TextMate for grammars because it's fast, and Monaco uses Monarch because TextMate doesn't work in web browsers.

registerLanguage() passes through to monaco.languages.register(), setLanguageConfiguration(), and setMonarchTokenProvider().

registerJsonSchemas() passes through to monaco.languages.json.jsonDefaults.setDiagnosticOptions.

onResourceChanged() is an aggregation of several similar events from VSCode/Monaco. 

IResource, getResourceUri, and getResourceVersion are all specific to the Azure DevOps code editor environment. They'd be roughly equivalent to the local file system module of VS Code.